src/lib/index.js

```diff
@@ -644,11 +643,7 @@ lib.minExtend = function(obj1, obj2) {
         if(k.charAt(0) === '_' || typeof v === 'function') continue;
         else if(k === 'module') objOut[k] = v;
         else if(Array.isArray(v)) {
-            if(k === 'colorscale') {
```

Comment:
Hmm. This probably slows down performance in some cases. 

Can we make try adding another special (similar to `k === 'colorscale'`) instead?

Reply:
Or perhaps a third arg to the function, to override `arrayLen = 3`, that gets invoked only when needed?

Reply:
> Or perhaps a third arg to the function, to override `arrayLen = 3`, that gets invoked only when needed?

Good call - that would work!

Reply:
Revised in c960d0d6c3e1005f143c30e05aa795cb477010a6.

Reply:
As `Lib.minExtend` is used in the toolpanel code, I would prefer if we would try to fix this bug without patching `Lib.minExtend`.

Sorry for the confusion @archmoj

Reply:
a third arg would be fine with toolpanel. I wouldn't want to add an exception like `color` because in some contexts that can be enormous. I would much prefer to pick out `pie` traces explicitly and extend the full length in that case, since we wouldn't be bothered by poor perf if someone gives us a pie with 100K slices 😏